themes/default/content/learn/abstraction-encapsulation/encapsulation/index.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "Understanding Encapsulation"
+layout: topic
+date: 2021-11-17
+draft: false
+description: |
+    Explore encapsulation of cloud resources with Pulumi.
+meta_desc: |
+    Explore encapsulation of cloud resources with Pulumi.
+index: 1
+estimated_time: 10
+meta_image: meta.png
+authors:
+    - laura-santamaria
+tags:
+    - learn
+block_external_search_index: false
+---
+
+So we've explored a bit about abstraction. What about encapsulation?
+
+Encapsulation is the logical conclusion of an abstracted system: Breaking logic
+down into reusable components. In doing so, we ensure better maintainability,
+readability, and reusability. To illustrate that, let's explore a practical
+example.
+
+## Breaking apart Pulumi code
+
+Let's take a fairly uncomplicated piece of Pulumi code: the definition of an s3
+bucket:
+
+```python
+import pulumi
+import pulumi_aws_native as aws
```

Comment:
Mmmmmmmaybe import `as aws_native` to make it glaringly obvious which provider is being used?

Reply:
Also consider just using AWS classic as referenced below to keep everything within a single provider, since AWS Native isn't going to be RTM until 2022.

Reply:
Good choice on switching it to `aws_native`. I've been just using `aws` in workshops and such, but I'm there to keep reminding the attendee of the import naming choice. Fixing in the next commit.

In regard to the question around the provider, we're trying to using the native provider whenever possible so folks get familiar with it. While I can see the argument, I think I'll stick with it as we have it. This part of the API is pretty stable as it's the one they're using to show off everything they're doing.